substrate-archive-backend/src/frontend.rs

```diff
@@ -91,12 +91,13 @@ fn default_block_workers() -> usize {
 	num_cpus::get()
 }
 
-impl From<RuntimeConfig> for ClientConfig {
-	fn from(config: RuntimeConfig) -> ClientConfig {
+impl<B: BlockT> From<RuntimeConfig> for ClientConfig<B> {
+	fn from(config: RuntimeConfig) -> ClientConfig<B> {
 		ClientConfig {
 			offchain_worker_enabled: false,
 			offchain_indexing_api: false,
 			wasm_runtime_overrides: config.wasm_runtime_overrides,
+			wasm_runtime_substitutes: Default::default(),
```

Comment:
This makes it compile I assume, but will it sync without the actual polkadot substitution when it gets to the infamous block?

Reply:
> Another question I have is how wasm_runtime_substitutes interacts with wasm_runtime_overrides. Which one should trump the other when they overlap?

Looking at the override code, code_substitute is used only if there isn't already an override passed in. So RuntimeOverrides trump code_substitute. 

To get over this hump when tracing is not enabled, however, will require some refactoring (working on that currently). Previously we didn't have to worry about the chain_spec at all when we could fully trust on-chain wasm. With `code_substitute` we will need to create ClientConfig out of the ChainSpec to sync past the problematic block. So yeah, `Default::default()` won't work in this case.